### PR TITLE
#83646 Webview: Cannot load resource outside of protocol root", if .md source is an network share

### LINKED ISSUE #83646: "Webview: Cannot load resource outside of protocol root", if .md source is an network share
if a markdown file is located on a network share, images `test` can not be loaded in preview.
Same folder on the local filesystem work fine.
image

Console display the error

webviewProtocols.ts:27 Webview: Cannot load resource outside of protocol root
e.session.protocol.registerBufferProtocol | @ | webviewProtocols.ts:27
-- | -- | --
  | async function (async) |   |  
  | e.session.protocol.registerBufferProtocol | @ | webviewProtocols.ts:19
  | apply | @ | C:\Users\adbase2\App…acks-registry.js:47
  | handleMessage | @ | C:\Users\adbase2\App…r\api\remote.js:285
  | ipcRenderer.on.args | @ | C:\Users\adbase2\App…r\api\remote.js:275
  | emit | @ | events.js:182

Version: 1.39.2 (user setup)
Commit: 6ab598523be7a800d7f3eb4d92d7ab9a66069390
Date: 2019-10-15T15:35:18.241Z
Electron: 4.2.10
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 10.0.17134

Steps to Reproduce:

1. create a markdown file with images on a network share.
2. markdown file and image are in the same notwork folder.
3. open preview

Does this issue occur when all extensions are disabled?: Yes

### PR DESCRIPTION
**Pull request by hadzhiyski**
ElectronWebviewBasedWebview local resource roots are with uppercase authority (host name)

This PR fixes #83646

### CODE REVIEW

**Comment by hadzhiyski** [comment]
Hello, @mjbvz ,

I am quite new to the vscode repository. Could you give me any guidelines if I am missing something? In my opinion the issue is fixed with latest changes.

File: src/vs/workbench/contrib/webview/common/resourceLoader.ts [hunk 1]
```diff
@@ -94,7 +95,18 @@ function normalizeRequestPath(requestUri: URI) {
...
 
-function containsResource(root: URI, resource: URI): boolean {
-	const rootPath = root.fsPath + (endsWith(root.fsPath, sep) ? '' : sep);
+function containsResource(root: URI, resource: URI, extensionLocation: URI | undefined): boolean {
+	let rootPath = root.fsPath + (endsWith(root.fsPath, sep) ? '' : sep);
+	let resourceFsPath = resource.fsPath;
+
+	// windows paths are case insensitive
+	// when checking remote host, the host name is uppercase
+	if (extensionLocation && (
```
Comment (mjbvz) [inline src/vs/workbench/contrib/webview/common/resourceLoader.ts]: Are you sure this is the correct fix? This code only applies if the extension is running remotely (which seems unrelated to #83646)

Comment (hadzhiyski) [reply src/vs/workbench/contrib/webview/common/resourceLoader.ts]: `isUNC(root.fsPath) && isUNC(resource.fsPath)` this should do.

**Comment by mjbvz** [comment]
Thanks! This fix will be in the next insiders build  and is scheduled to ship with VS Code 1.41 early next month

**Comment by hadzhiyski** [comment]
Thank you for the support!

**Pull request merged**